src/attached_probe.cpp

```diff
@@ -842,6 +842,12 @@ void AttachedProbe::load_prog(BPFfeature &feature)
 
       if (opts.attach_prog_fd > 0)
         close(opts.attach_prog_fd);
+      if (btf_fd_ >= 0) {
+        close(btf_fd_);
+        btf_fd_ = -1;
+      }
```

Comment:
How about moving this to a local variable? Seems like `btf_fd_` is unused outside this function, save for closing in destructor.

Reply:
Thanks @jordalgo and @danobi for the review.

Daniel, you are absolutely correct and that's me being lazy - I'll convert it to a local variable as there's no need for it to be a class member.

Re: formatting. In my defence ( :-) ) I was using the signal I got from my own branch before I created the PR and that was all green. @ajor just looked and formatting signal is only provided on PR's and not for pushes to private branches. Could a clang-format check be added to the branch pushes as this would stop thsi kind off issue?

Reply:
Derailing the PR a little, but specifically this part of the CI config: https://github.com/iovisor/bpftrace/blob/20944038eef747e3c6255edb178db66bbd72737e/.github/workflows/ci.yml#L11-L14

I don't understand what the purpose is - can we remove it?

Reply:
I think if you take that out then the job will start running on master branch. Then you'd see failures if people chose to ignore clang-format. If we are ok with making it a strict requirement (cuz in practice it is), we can delete the gate.

Reply:
I think it'd allow bad formatting on master anyway, because it only formats the diff between the current commit and master (which would have zero lines when on master). Opened: https://github.com/iovisor/bpftrace/pull/2931